src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1481,13 +1481,57 @@ private function registerAnnotationsConfiguration(array $config, ContainerBuilde
                 ->setMethodCalls([['registerLoader', ['class_exists']]]);
         }
 
-        if ('none' !== $config['cache']) {
+        if ('none' === $config['cache']) {
+            $container->removeDefinition('annotations.cached_reader');
+
+            return;
+        }
+
+        $cacheService = $config['cache'];
+        if (\in_array($config['cache'], ['php_array', 'file'])) {
+            $isPsr6Service = $container->hasDefinition('annotations.psr_cached_reader');
+        } else {
+            $isPsr6Service = is_subclass_of($container->getDefinition($cacheService)->getClass(), CacheItemPoolInterface::class);
```

Comment:
I fear this won't work because the service might not be loaded yet.
Instead, I propose to deprecate the option, and wire a psr6 only when this is empty (aka only the default cache is psr6)
There is no need to make this configurable IMHO.